unit-test-security-rules-v9/test/firestore.spec.mjs

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import assert from 'assert';
+import { readFileSync } from 'fs';
+
+// Workarounds for npm not recognizing named exports in cjs.
+import appNs from 'firebase/app';
+const { initializeApp, deleteApp, getApps } = appNs;
+
+// We can use firestore/lite since we don't need persistence or use listeners.
+// Feel free to switch to firebase/firestore if your tests use onSnapshot etc.
+import firestoreNs from 'firebase/firestore/lite';
+const { getFirestore, useFirestoreEmulator, doc, getDoc, setDoc, serverTimestamp, setLogLevel } = firestoreNs;
+
+import adminAppNs from 'firebase-admin/app';
+const { initializeApp: initializeAdminApp } = adminAppNs;
+import adminFirestoreNs from 'firebase-admin/firestore';
+const { getFirestore: getAdminFirestore } = adminFirestoreNs;
+import adminSecurityRulesNs from 'firebase-admin/security-rules';
+const { getSecurityRules } = adminSecurityRulesNs;
+
+/**
+ * The emulator will accept any project ID for testing.
```

Comment:
As long as it is lowercase? https://cloud.google.com/resource-manager/reference/rest/v1beta1/projects